packages/instrumentation-sequelize/src/types.ts

```diff
@@ -1,9 +1,20 @@
 import { Span } from '@opentelemetry/api';
 import { InstrumentationConfig } from '@opentelemetry/instrumentation';
 
+export interface SequelizeQueryHookParams {
+  /** The type of sql parameter depends on the database dialect. */
```

Comment:
Hmm... this statement probably needs validating.

Reply:
I'm ok with type `any`, but to automatically extract the exact type (which will allow typescript to perform type validations), we can write this:
```js
import type { Sequelize } from 'sequelize';
export interface SequelizeQueryHookParams {
  sql: Parameters<Sequelize['query']>[0],
  option: Parameters<Sequelize['query']>[1],
};
```

An alternative will be to copy the relevant types by [following sequalize source code](https://github.com/sequelize/sequelize/blob/56bb1d6ef9827f604d7bcef945abb7e213f2322d/types/lib/sequelize.d.ts#L1206):
```js
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.UPDATE>): Promise<[undefined, number]>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.BULKUPDATE>): Promise<number>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.INSERT>): Promise<[number, number]>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.UPSERT>): Promise<number>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.DELETE>): Promise<void>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.BULKDELETE>): Promise<number>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.SHOWTABLES>): Promise<string[]>;
  public query(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.DESCRIBE>): Promise<ColumnsDescription>;
  public query<M extends Model>(
    sql: string | { query: string; values: unknown[] },
    options: QueryOptionsWithModel<M>
  ): Promise<M[]>;
  public query<T extends object>(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.SELECT> & { plain: true }): Promise<T>;
  public query<T extends object>(sql: string | { query: string; values: unknown[] }, options: QueryOptionsWithType<QueryTypes.SELECT>): Promise<T[]>;
  public query(sql: string | { query: string; values: unknown[] }, options: (QueryOptions | QueryOptionsWithType<QueryTypes.RAW>) & { plain: true }): Promise<{ [key: string]: unknown }>;
  public query(sql: string | { query: string; values: unknown[] }, options?: QueryOptions | QueryOptionsWithType<QueryTypes.RAW>): Promise<[unknown[], unknown]>;
```

So for the `sql` parameter, the type is `string | { query: string; values: unknown[] }`, and for the `options`, the type is `QueryOptionsWithType` with few `QueryTypes` options.

I am OK with all options so feel free to choose the one you best feel suited here.

Reply:
I tried the `Parameters<Sequelize['query']>` approach and it seems to work well, so I'll go with that. (Interestingly, the value I want to add to the span `sql.bind` isn't specified in that type.)